lib/checks/forms/group-labelledby.js

```diff
@@ -1,34 +1,45 @@
+const { utils, dom, text } = axe.commons;
+
 this.data({
 	name: node.getAttribute('name'),
 	type: node.getAttribute('type')
 });
 
-var doc = axe.commons.dom.getRootNode(node);
-var matchingNodes = doc.querySelectorAll(
-	'input[type="' +
-		axe.commons.utils.escapeSelector(node.type) +
-		'"][name="' +
-		axe.commons.utils.escapeSelector(node.name) +
-		'"]'
+const doc = dom.getRootNode(node);
+const matchingNodes = Array.from(
+	doc.querySelectorAll(
+		'input[type="' +
+			utils.escapeSelector(node.type) +
+			'"][name="' +
+			utils.escapeSelector(node.name) +
+			'"]'
+	)
 );
 if (matchingNodes.length <= 1) {
 	return true;
 }
 
 // Check to see if there's an aria-labelledby value that all nodes have in common
-return (
-	[].map
-		.call(matchingNodes, function(m) {
-			var l = m.getAttribute('aria-labelledby');
-			return l ? l.split(/\s+/) : [];
-		})
-		.reduce(function(prev, curr) {
-			return prev.filter(function(n) {
-				return curr.includes(n);
-			});
-		})
-		.filter(function(n) {
-			var labelNode = doc.getElementById(n);
-			return labelNode && axe.commons.text.accessibleText(labelNode, true);
-		}).length !== 0
-);
+const sharedIds = matchingNodes
+	.map(m => {
+		return utils.tokenList(m.getAttribute('aria-labelledby') || '');
+	})
+	.reduce((prev, curr) => {
```

Comment:
Although `reduce` can be called with out an `initialValue | Optional`, in which case, the first value in the `array` is used as `initialValue`, this will cause an error/ break when the above `map` returns an empty array, as the below line `prev.filter` will fail as `prev` will be `undefined`.